src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5919,6 +5917,34 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       if (phase.erasedTypes) None // OPT save the hashmap lookup in erasure type and beyond
       else transformed remove tree
   }
+
+
+  /** Finish computation of param aliases after typechecking is completed */
```

Comment:
It would be nice to have a (very short) summary what happens with param aliases (or say where it's documented)...

Reply:
I've added a few comments with my best reverse engineering of the intent.